Sprint-1/errors/3.js

```diff
@@ -1,9 +1,17 @@
-const cardNumber = 4533787178994213;
-const last4Digits = cardNumber.slice(-4);
+//const cardNumber = 4533787178994213;
+//const last4Digits = cardNumber.slice(-4);
 
 // The last4Digits variable should store the last 4 digits of cardNumber
 // However, the code isn't working
 // Before running the code, make and explain a prediction about why the code won't work
 // Then run the code and see what error it gives.
 // Consider: Why does it give this error? Is this what I predicted? If not, what's different?
 // Then try updating the expression last4Digits is assigned to, in order to get the correct value
+
+///==> my prediction: it won't work because carNumber is neither an array nor a string which means we cannot get the index the way we are doing now
+// One way to solve this is:
+// 1. Convert carNumber into a string
+// 2. subtracts 4 to the length of cardNumber to be able to extract the last 4 elements since we not dealing with and array
+const cardNumber = "4533787178994213";
+const last4Digits = +cardNumber.slice(cardNumber.length- 4);
```

Comment:
This looks good and works, but can you think of a way of doing this which _doesn't_ require you to change how the `cardNumber` variable was declared?

i.e. so that the first line is still `const cardNumber = 4533787178994213;`, but `last4Digits` still ends up being a number containing `4213`?

Reply:
Yes, I think I could use the toString() method to convert the variable into a string first instead of doing it manually. Like the following
```
const cardNumber = 4533787178994213;
const last4Digits = +cardNumber.toString().slice(- 4);
```